configure.ac

```diff
@@ -350,31 +350,81 @@ AM_CONDITIONAL([ENABLE_GRPTEST_LDMS_TEST], [test "x$ENABLE_LDMS_TEST_FALSE" = "x
-    AC_CHECK_HEADER(jansson.h, [],                                              
-        AC_MSG_ERROR([`jansson.h` not found (required by variorum_ldms). You may need to set JANSSON_DIR if jansson is already installed.]))
-fi                                                                              
+dnl OPTION_DEFAULT_ENABLE([variorum_ldms],[ENABLE_VARIORUM_LDMS])
+
+dnl Variorum dependency support
+AC_MSG_CHECKING([for Variorum sampler])
+AC_ARG_WITH([variorum],
+        [AS_HELP_STRING([--with-variorum],
+                [build Variorum sampler @<:@default=check@:>@])],
+                [AS_IF([test "x$withval" != xyes -a "x$withval" != xno],
+                        [VARIORUM_CFLAGS="-I$withval/include"
+                         VARIORUM_LDFLAGS="-L$withval/lib"]
+                )],
+        [with_variorum=check])
+have_variorum=no
+AC_ARG_WITH([jansson],
+        [AS_HELP_STRING([--with-jansson],
+                [include Jansson library for Variorum sampler @<:@default=check@:>@])],
+                [AS_IF([test "x$withval" != xyes -a "x$withval" != xno],
+                        [JANSSON_CFLAGS="-I$withval/include"
+                         JANSSON_LDFLAGS="-L$withval/lib"]
+                )],
+        [with_jansson=check])
+AC_SUBST([VARIORUM_CFLAGS])
+AC_SUBST([VARIORUM_LDFLAGS])
+AC_SUBST([JANSSON_CFLAGS])
+AC_SUBST([JANSSON_LDFLAGS])
+AS_IF([test "x$with_variorum" != xno], [
+        save_LDFLAGS=$LDFLAGS
+        LDFLAGS="$LDFLAGS $VARIORUM_LDFLAGS $JANSSON_LDFLAGS"
+        save_CFLAGS=$CFLAGS
+        CFLAGS="$CFLAGS $VARIORUM_CFLAGS $JANSSON_CFLAGS"
+        save_CPPFLAGS=$CPPFLAGS
+        CPPFLAGS="$CPPFLAGS $VARIORUM_CFLAGS $JANSSON_CFLAGS"
+        AC_CHECK_HEADER(variorum.h, [have_variorum=yes],[have_variorum=no]
+          AC_MSG_ERROR([`variorum.h` not found (required by variorum_ldms). You may need to set VARIORUM_DIR if variorum is already installed.]))
+        AC_CHECK_HEADER(variorum_topology.h, [have_variorum=yes],[have_variorum=no]
+          AC_MSG_ERROR([`variorum_topology.h` not found (required by variorum_ldms). You may need to update variorum.]))
+        AC_CHECK_HEADER(jansson.h, [have_variorum=yes],[have_variorum=no]
+          AC_MSG_ERROR([`jansson.h` not found (required by variorum_ldms). You may need to set JANSSON_DIR if jansson is already installed.]))
+        AC_CHECK_LIB(jansson, json_real_value, [have_variorum=yes],[have_variorum=no]
+          AC_MSG_ERROR([jansson not found (required by variorum_ldms). You may need to set JANSSON_DIR if jansson is already installed.]))
+        AC_CHECK_LIB(variorum, variorum_get_node_power_json, [have_variorum=yes],[have_variorum=no]
+          AC_MSG_ERROR([variorum not found (required by variorum_ldms). You may need to set VARIORUM_DIR if variorum is already installed.]))
+        CPPFLAGS=$save_CPPFLAGS
+        CFLAGS=$save_CFLAGS
+        LDFLAGS=$save_LDFLAGS
+])
+AM_CONDITIONAL([ENABLE_VARIORUM], [test x$have_variorum = xyes])
```

Comment:
We only want to set the ENABLE_VARIORUM AM conditional is _both_ have_variorum is yes and have_jansson is yes. (You'll need to add the test for libjansson as I mentioned, and replicated most of the comments I have mad about the variorum section in the jansson section.)